src/Handlers/Rules.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace AntispamBee\Handlers;
+
+use AntispamBee\Helpers\ComponentsHelper;
+use AntispamBee\Interfaces\Controllable;
+use AntispamBee\Interfaces\SpamReason;
+use AntispamBee\Interfaces\Verifiable;
+
+class Rules {
+	protected $type;
+	protected $spam_reasons = [];
+	protected $no_spam_reasons = [];
+
+	public function __construct( $type ) {
+		$this->type = $type;
+	}
+
+	public function apply( $item ) {
+		$item['reaction_type'] = $this->type;
+		$rules                = self::get( $this->type, true );
+
+		$score = 0.0;
+		foreach ( $rules as $rule ) {
+			$rule_score = $rule::verify( $item ) * $rule::get_weight();
+
+			if ( $rule_score > 0.0 ) {
+				$this->spam_reasons[] = $rule::get_slug();
+			} else {
+				$this->no_spam_reasons[] = $rule::get_slug();
+			}
+
+			$score += $rule_score;
+
+			$no_spam_threshold = (float) apply_filters( 'antispam_bee_no_spam_threshold', 0.0 );
+			$spam_threshold    = (float) apply_filters( 'antispam_bee_spam_threshold', 0.0 );
+
+			if ( $no_spam_threshold < 0.0 && $score <= $no_spam_threshold ) {
```

Comment:
Move the threshold checks below the `foreach`. Run the spam threshold check before the no_spam threshold check